data_diff/databases/presto.py

```diff
@@ -38,7 +37,25 @@ class Presto(Database):
     def __init__(self, **kw):
         prestodb = import_presto()
 
-        self._conn = prestodb.dbapi.connect(**kw)
+        if kw.get("schema"):
+            self.default_schema = kw.get("schema")
+
+        try:
+            # checks if user and password are missing when auth=basic
+            kw.get("auth") == "basic" and "user" in kw and "password" in kw
```

Comment:
This doesn't do anything, won't even throw an exception.

Reply:
It throws an exception if user or password is missing. I tested it

This is after replacing `ValueError` with `KeyError`:
```
try:
            # checks if user and password are missing when auth=basic
            kw.get("auth") == "basic" and "user" in kw and "password" in kw
            # if auth=basic, add basic authenticator for Presto
            kw["auth"] = prestodb.auth.BasicAuthentication(kw.pop("user"), kw.pop("password"))
        except:
            raise KeyError("User or password cannot be missing if auth==basic")
```

Reply:
> It throws an exception if user or password is missing. I tested it

The second line does. The first line doesn't do anything.

Reply:
Do you mean having the code like this instead:
```
if kw.get("auth") == "basic":
            kw["auth"] = prestodb.auth.BasicAuthentication(kw.pop("user"), kw.pop("password"))
```
So that it throws error if user or password is missing?

Reply:
Yes, that's more inline with what I'm thinking.

Reply:
Thanks! I have corrected that now